### ISSUE TITLE
Failed reading /var/run/secrets/kubernetes.io/serviceaccount/token

### ISSUE THREAD

**Issue by fabio-rigato** [minherz/gke_node_labeling#1]
Hello,

I'm facing an issue running the node-labeling process.

the pod in not running properly and this is the log:

```
Installing collected packages: six, pyasn1, urllib3, rsa, pyparsing, pyasn1-modules, protobuf, idna, chardet, certifi, cachetools, requests, pytz, packaging, httplib2, googleapis-common-protos, google-auth, uritemplate, google-auth-httplib2, google-api-core, google-api-python-client
Successfully installed cachetools-4.2.2 certifi-2021.5.30 chardet-4.0.0 google-api-core-1.27.0 google-api-python-client-1.12.8 google-auth-1.24.0 google-auth-httplib2-0.0.4 googleapis-common-protos-1.53.0 httplib2-0.19.1 idna-2.10 packaging-20.9 protobuf-3.17.2 pyasn1-0.4.8 pyasn1-modules-0.2.8 pyparsing-2.4.7 pytz-2021.1 requests-2.25.1 rsa-4.7.2 six-1.16.0 uritemplate-3.0.1 urllib3-1.26.5
WARNING: Running pip as root will break packages and permissions. You should install packages reliably by using venv: https://pip.pypa.io/warnings/venv
Failed reading /var/run/secrets/kubernetes.io/serviceaccount/token
```

I'm running on GKE with workload identity, I created a Cloud SA with the role 
```roles/compute.instanceAdmin```
and bounded with a k8s SA with the same ClusterRole.

Where I am wrong?

thank you so much in advance, for your help and effort.

With all the best,
Fabio Rigato

**Comment by minherz** [comment]
The reason is probably in workload identity enabled on the cluster. I will have to check it but this is my first guess. I did not check the solution on the clusters with workload identity enabled.
Let me to check it and get back to you.

**Comment by fabio-rigato** [comment]
Thanks, @minherz for your fast reply.

I was able to solve the problem, I was creating the SA with automountServiceAccountToken: false
here for more info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server

In particular, since I'm using workload identity I had to:

1. create the k8s SA and bind it with the IAM SA
2. Create a ClusterRoleBinding associated to node-info-reader ClusterRole
3. Update the serviceAccountName with me new SA in the Deamonset

Thanks for your time anyway.

With all the best,
Fabio Rigato
